src/Workspaces/Core/Portable/Workspace/Solution/Document.cs

```diff
@@ -154,18 +154,8 @@ public bool SupportsSemanticModel
             }
         }
 
-        /// <summary>
-        /// Gets the <see cref="SyntaxTree" /> for this document asynchronously.
-        /// </summary>
-        public Task<SyntaxTree> GetSyntaxTreeAsync(CancellationToken cancellationToken = default(CancellationToken))
+        private Task<SyntaxTree> GetExistingSyntaxTreeTask()
```

Comment:
GetExistingCachedSyntaxTreeTask? It took me awhile to figure out what the difference was between this and the regular Async function.

(I'm also terribly scared that we're caching syntax trees like this. So we don't hold the tree alive unless a second call comes, and then it might work and be pinned forever?)